### ISSUE TITLE
nsfw_reddit dataset details

### ISSUE THREAD

**Issue by maksimstw** [acheong08/acheong08#8]
Hi acheong08! I am a CS PhD student at USC working on alignment and LLMs safety. I recently found your [nsfw_reddit](https://huggingface.co/datasets/acheong08/nsfw_reddit) dataset. Would you mind sharing how you collected such a dataset? It would be quite valuable to our research. Thank you!

**Comment by acheong08** [comment]
Hahahahaha... Embarrassing that someone found that. I was trying to train a "bad" LLM

It came from Reddit's API back when it was still available. [It is now more limited s](https://en.wikipedia.org/wiki/2023_Reddit_API_controversy) so I'm not sure how much data you can get. It's unfortunate that I didn't get more data when I could.

Try https://github.com/praw-dev/praw.

I seem to have deleted the source code used to get my dataset so I apologize for not being able to help more.

**Comment by maksimstw** [comment]
Thanks! I've seen people using data from 4Chan to train a "bad" LLM ([gpt-4chan](https://huggingface.co/ykilcher/gpt-4chan)). You might want to check it out if you are interested.

Could you please elaborate a bit more on how you filter the data though? What kind of heuristic did you use? Did you do keyword/flair matching? And also what subreddit did you explore?

**Comment by acheong08** [comment]
> Could you please elaborate a bit more on how you filter the data though? What kind of heuristic did you use? Did you do keyword/flair matching? And also what subreddit did you explore?

The data was not filtered. It was a simple crawl and dump.

https://www.reddit.com/r/ListOfSubreddits/wiki/nsfw/

Posts with images were removed. Most data come from r/*stories

**Comment by maksimstw** [comment]
Thanks for the explanation! Closing the issue now.

**Comment by maksimstw** [comment]
I am wondering if you could provide the time period you collected such a dataset. Also, is there a way I could cite the dataset? If you don't mind, shoot me an email at taiweish@usc.edu and perhaps we could communicate on WeChat as well.

**Comment by acheong08** [comment]
Data was collected within a day on March 15th 2023. It was initially created with subreddits I got just by searching Reddit. Update on April 9th added more data sourced from subreddits in the [list](https://www.reddit.com/r/ListOfSubreddits/wiki/nsfw/). It wasn't academic so I don't really know how you would cite it. Perhaps just with a link or something.

You might want to take a look at https://huggingface.co/datasets/jjmachan/NSFW-reddit as it has much more data than what I was able to scrape.
